src/engine/LoadURL.cpp

```diff
@@ -0,0 +1,147 @@
+size_t LoadURL::getResultWidth() const { return 3; }
+
+// _____________________________________________________________________________
+size_t LoadURL::getCostEstimate() {
+  // TODO: For now, we don't have any information about the cost at query
+  // planning time, so we just return ten times the estimated size.
+  return 10 * getSizeEstimateBeforeLimit();
+}
+
+// _____________________________________________________________________________
+uint64_t LoadURL::getSizeEstimateBeforeLimit() {
+  // TODO: For now, we don't have any information about the result size at
+  // query planning time, so we just return `100'000`.
+  return 100'000;
+}
+
+// _____________________________________________________________________________
+float LoadURL::getMultiplicity(size_t) {
+  // TODO: For now, we don't have any information about the multiplicities at
+  // query planning time, so we just return `1` for each column.
+  return 1;
+}
+
+// _____________________________________________________________________________
+bool LoadURL::knownEmptyResult() { return false; }
+
+// _____________________________________________________________________________
+std::unique_ptr<Operation> LoadURL::cloneImpl() const {
+  auto load = std::make_unique<LoadURL>(_executionContext, loadURLClause_);
+  load->cacheBreaker_ = cacheBreaker_;
+  return load;
+}
+
+// _____________________________________________________________________________
+vector<ColumnIndex> LoadURL::resultSortedOn() const { return {}; }
+
+// _____________________________________________________________________________
+Result LoadURL::computeResult(bool) {
+  // TODO<qup42> implement lazy loading; requires modifications to the parser
+  LOG(INFO) << "Loading RDF dataset from " << loadURLClause_.url_.asString()
+            << std::endl;
+  HttpOrHttpsResponse response =
+      getResultFunction_(loadURLClause_.url_, cancellationHandle_,
+                         boost::beast::http::verb::get, "", "", "");
+
+  auto throwErrorWithContext = [this, &response](std::string_view sv) {
+    std::string ctx;
+    ctx.reserve(100);
+    for (const auto& bytes : std::move(response.body_)) {
+      ctx += std::string(reinterpret_cast<const char*>(bytes.data()),
+                         bytes.size());
+      if (ctx.size() >= 100) {
+        break;
+      }
+    }
+    this->throwErrorWithContext(sv, std::string_view(ctx).substr(0, 100));
+  };
+
+  if (response.status_ != boost::beast::http::status::ok) {
+    throwErrorWithContext(absl::StrCat(
+        "RDF dataset responded with HTTP status code: ",
+        static_cast<int>(response.status_), ", ",
+        toStd(boost::beast::http::obsolete_reason(response.status_))));
+  }
+  std::optional<ad_utility::MediaType> mediaType =
+      ad_utility::toMediaType(response.contentType_);
+  if (!mediaType) {
+    throwErrorWithContext(
+        "QLever requires the endpoint of a LoadURL to return the mediatype.");
```

Comment:
1. What happens in `toMediaType` for a garbage media type that cannot even be converted?
In this case we should have a proper error message. 
The current message uses `mediatype` which is not a thing, but the response must have the `content-header` set to a supported value, and then it is either empty, or something unsupported which should be echoed to the user.